lib/rules/no-eq-neg-zero.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @fileoverview The rule should warn against code that tries to use the === operator to compare against -0.
+ * @author no-eq-neg-zero
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow use the === operator to compare against -0.",
+            category: "Possible Errors",
+            recommended: true
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+        return {
+            BinaryExpression(node) {
+                if (/===?$/.test(node.operator) && (node.right.operator === "-") && node.right.argument.value === 0) {
+                    context.report({ node, message: "disallow use the === operator to compare against -0." });
```

Comment:
I agree, and think the rule should be expanded to cover all comparison operators, since e.g. `(0 > -0) === false` is surprising in exactly the same way as `(0 === -0) === true` and `(0 !== -0) === false`.